### ISSUE TITLE
Incorrect Tab order for popover of DateInput2

### ISSUE THREAD

**Issue by seb5s** [palantir/blueprint#6034]
#### Environment

- __Package version(s)__: @blueprintjs/datetime2: 0.9.24
- __Operating System__: mac OS
- __Browser name and version__:  I can replicate the bug in the latest versions of Chrome, Edge and Safari.

#### Code Sandbox

 [Link to a minimal repro](https://codesandbox.io/s/ancient-fast-53ruuy?file=/src/DatetimeExample.tsx)

#### Actual behavior

When you have focus in the DateInput2 field and the popover is open and you click the `tab` keyboard key, you are moved to the next input field (not into the date selector in the popover).

The bug makes it hard to fill out forms only using the keyboard and poses accessibility issues. Focus can be moved to an input field behind the popover which is confusing.

https://user-images.githubusercontent.com/12937146/226369552-12524037-f318-4005-b6b0-cd9150e8484e.mov

#### Expected behavior

I would expect behavior similar to the `DateInput` (see beginning of the video). When you have focus in the input field and the popover is open and you click the `tab` keyboard key I would expect focus to be put on the first button on the date selector popover.

#### Possible solution

Update tab order, so that the inputs in the data selector is not put in the end.

**Comment by adidahiya** [comment]
Thanks for flagging this issue. This certainly seems like a regression from DateInput, likely caused by our migration to Popover2. I'll try to fix it soon, but I'm also open to PRs in the meantime.

I fixed a couple syntax / TypeScript errors in your code sandbox: https://codesandbox.io/s/focused-snowflake-l7985e?file=/src/DatetimeExample.tsx

**Comment by camposfyi** [comment]
I'm experiencing the same issue.

**Comment by camposfyi** [comment]
I am experiencing the same issue. In my case I don't open the calendar when the user clicks the input. I have a calendar icon that triggers the popover to open.
<img width="529" alt="image" src="https://github.com/palantir/blueprint/assets/5265784/a550c8ad-0b29-4d78-b46a-82a9293acb04">

In DateTimeInput.tsx when the input is focused, the internal state gets set as open:

```
    const handleInputFocus = React.useCallback(
        (e: React.FocusEvent<HTMLInputElement>) => {
            setIsInputFocused(true);
            setIsOpen(true);
            setInputValue(formattedDateString);
            props.inputProps?.onFocus?.(e);
        },
        [formattedDateString, props.inputProps?.onFocus],
    );
```

but in my component I am controlling when the popover should be open

```
            popoverProps={{
                position: 'bottom',
                isOpen: selectedCalendar === 'startDate',
                onInteraction: onCalendarPopoverInteraction('startDate'),
            }}
```

If I click the date input and the onFocus is called, and then try to tab out of it, it gets stuck because of this piece of code, notice that `getKeyboardFocusableElements` will return an empty array because the popover is closed, and then it calls `e.preventDefault()` 
```
if (popoverContentRef.current === null) {
        return [];
}
```

```
    const handleInputKeyDown = React.useCallback(
        (e: React.KeyboardEvent<HTMLInputElement>) => {
            if (e.key === "Tab" && e.shiftKey) {
                // close popover on SHIFT+TAB key press
                handlePopoverClose(e);
            } 
           // <<<<<<<<<<<------------- THIS RIGHT HERE ---------------->>>>>>>>>>>>>>>>
           else if (e.key === "Tab" && isOpen) {
                getKeyboardFocusableElements(popoverContentRef).shift()?.focus();
                // necessary to prevent focusing the second focusable element
                e.preventDefault();
            } else if (e.key === "Escape") {
                setIsOpen(false);
                inputRef.current?.blur();
            } else if (e.key === "Enter" && inputValue != null) {
                const nextDate = parseDate(inputValue);
                if (isDateValid(nextDate)) {
                    handleDateChange(nextDate, true, true);
                }
            }

            props.inputProps?.onKeyDown?.(e);
        },
        [inputValue, parseDate, props.inputProps?.onKeyDown],
    );
```

**Comment by evansjohnson** [comment]
This appears to be due to `flex-direction: row-reverse;` styling, which modifies visual order but does not change tab order. Unfortunately it seems like we don't have options to actually move the DOM elements of the underlying library, so I don't see an easy way to fix this. The underlying library does not seem to have a built in option to order these navigation elements.

https://github.com/palantir/blueprint/blob/0d36ba1cae2c5c44f77efe1bf5e6cb055cf2f959/packages/datetime2/src/components/date-range-picker3/_date-range-picker3.scss#L32

https://github.com/palantir/blueprint/blob/0d36ba1cae2c5c44f77efe1bf5e6cb055cf2f959/packages/datetime2/src/components/date-picker3/_date-picker3.scss#L112

Note it seems like the date input component has changed since filing, to put the arrows together, possibly to improve this. But this behavior is still present when reversing month and year dropdowns, and when using two dateinputs next to eachother in the daterange component

**Comment by evansjohnson** [comment]
@camposfyi that sounds like a different issue, this is for the tab order and it sounds like you're talking about focus being trapped or lost. Could you open a new issue with a reproducer?
